### PR TITLE
add try/catch to calls to PerformanceObserver.prototype.observe and add tests

### PR DESCRIPTION
**Pull request by erwinmombay**
the performance tracking instance is initialized very early in the amp initialization lifecycle. A throw to a call to `observe` can occur and break the initialization of the whole runtime. Make it more resilient to this case by wrapping all calls to observe in a try catch block

### CODE REVIEW

File: src/service/performance-impl.js [hunk 1]
```diff
@@ -420,8 +441,7 @@ export class Performance {
         );
         navigationObserver.observe({type: 'navigation', buffered: true});
       } catch (err) {
-        dev() /*OK*/
-          .error(err);
+        dev()./*OK*/ error(err);
```
Comment (jridgewell) [inline src/service/performance-impl.js:444]: The warning here is that you're supposed to passing a `TAG`: `dev().error(TAG, err)`

File: src/service/performance-impl.js [hunk 2]
```diff
@@ -387,28 +387,49 @@ export class Performance {
...
+      }
     }
 
     if (this.supportsLargestContentfulPaint_) {
-      const lcpObserver = this.createPerformanceObserver_(processEntry);
-      lcpObserver.observe({type: 'largest-contentful-paint', buffered: true});
+      try {
+        const lcpObserver = this.createPerformanceObserver_(processEntry);
+        lcpObserver.observe({type: 'largest-contentful-paint', buffered: true});
```
Comment (jridgewell) [inline src/service/performance-impl.js:429]: Nit: We can move the observe call into `createPerformanceObserver_` by passing this init object, which we can then reuse a single try-catch. That should reduce overall code size.

**Review by jridgewell — CHANGES_REQUESTED** [review]

File: src/service/performance-impl.js (hunk 2 above)
Comment (erwinmombay) [reply src/service/performance-impl.js]: done. moved the `try/catch` into the method.

File: src/service/performance-impl.js (hunk 1 above)
Comment (erwinmombay) [reply src/service/performance-impl.js]: done. updated the error report in the moved `try/catch`

**Review by samouri — APPROVED** [review]

File: src/service/performance-impl.js [hunk 3]
```diff
@@ -395,63 +395,59 @@ export class Performance {
...
+        type: 'navigation',
+        buffered: true,
+      });
     }
 
     if (entryTypesToObserve.length === 0) {
       return;
     }
 
     const observer = this.createPerformanceObserver_(processEntry);
```
Comment (jridgewell) [inline src/service/performance-impl.js:432]: We need an `init` here.

**Review by jridgewell — APPROVED** [review]
Pushed a new commit.

**Pull request merged**